modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/sort/SortParserImpl.java

```diff
@@ -74,8 +74,17 @@ public List<SortField> parse(String sortString) {
 			return Collections.emptyList();
 		}
 
-		return TransformUtil.transform(
-			StringUtil.split(sortString), string -> getSortField(string));
+		try {
+			return TransformUtil.transform(
+				StringUtil.split(sortString), string -> getSortField(string));
+		}
+		catch (RuntimeException runtimeException) {
+			if (runtimeException.getCause() instanceof InvalidSortException) {
+				throw (InvalidSortException)runtimeException.getCause();
+			}
+
+			throw runtimeException;
+		}
```

Comment:
I had to do this horrible code to get a InvalidSort, otherwise it was throwing a runtime exception :'(

Reply:
See https://github.com/liferay-headless/liferay-portal/pull/1337.